kyuubi-common/src/main/scala/org/apache/kyuubi/util/KyuubiHadoopUtils.scala

```diff
@@ -33,4 +35,24 @@ object KyuubiHadoopUtils {
   def getServerPrincipal(principal: String): String = {
     SecurityUtil.getServerPrincipal(principal, "0.0.0.0")
   }
+
+  def encodeCredentials(creds: Credentials): String = {
```

Comment:
```scala
  def serialize(creds: Credentials): Array[Byte] = {
    val byteStream = new ByteArrayOutputStream
    val dataStream = new DataOutputStream(byteStream)
    creds.writeTokenStorageToStream(dataStream)
    byteStream.toByteArray
  }

  def deserialize(tokenBytes: Array[Byte]): Credentials = {
    val tokensBuf = new ByteArrayInputStream(tokenBytes)

    val creds = new Credentials()
    creds.readTokenStorageStream(new DataInputStream(tokensBuf))
    creds
  }

```

the implementation in Spark side looks more elegant to me

Reply:
we can store bytes in memory directly?

Reply:
Use Spark's implementation since it is simpler. 
**Note:** `ByteArrayOutputStream` also involves  an array copy action.